### PR TITLE
Rewrite some of the AD tests with Tracked<Float>

### PR DESCRIPTION
**Pull request by bgogul**
First of the several PRs that will enable leak checking for the AD tests. 

https://bugs.swift.org/browse/TF-895

### CODE REVIEW

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

**Review by dan-zheng — APPROVED** [review]
Great!

File: stdlib/private/DifferentiationUnittest/DifferentiationUnittest.swift [hunk 1]
```diff
@@ -257,6 +257,34 @@ extension Tracked where T : Differentiable & FloatingPoint, T == T.TangentVector
   }
 }
 
+extension Tracked : VectorProtocol where T : SignedNumeric, T == T.Magnitude {
```
Comment (rxwei) [inline stdlib/private/DifferentiationUnittest/DifferentiationUnittest.swift]: What test is using this conformance? VectorProtocol is not directly related to AD and cannot be upstreamed to master.

**Review by rxwei — APPROVED** [review]

File: stdlib/private/DifferentiationUnittest/DifferentiationUnittest.swift (hunk 1 above)
Comment (dan-zheng) [reply stdlib/private/DifferentiationUnittest/DifferentiationUnittest.swift]: Some `AutoDiff` tests reference `VectorProtocol` for various reasons, I think all such references can be removed.

Comment (rxwei) [reply stdlib/private/DifferentiationUnittest/DifferentiationUnittest.swift]: Let’s do that then.

Comment (bgogul) [reply stdlib/private/DifferentiationUnittest/DifferentiationUnittest.swift]: This was needed for the protocol_requirement_autodiff.swift test. If we don't need `VectorProtocol`, I will go ahead and clean that up as well.

Comment (rxwei) [reply stdlib/private/DifferentiationUnittest/DifferentiationUnittest.swift]: You can try changing whatever requires VectorProtocol to require AdditiveArithmetic instead.

Comment (bgogul) [reply stdlib/private/DifferentiationUnittest/DifferentiationUnittest.swift]: Removed VectorProtocol here: https://github.com/apple/swift/pull/27734 

Once that gets submitted, I will pull in the relevant changes.

Comment (bgogul) [reply stdlib/private/DifferentiationUnittest/DifferentiationUnittest.swift]: I reverted the changes to  `DifferentiationUnittest.swift`

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Pull request merged**